brave/brave-core
# Use different icon for some file types when brave is set as a default application

fix https://github.com/brave/brave-browser/issues/12761

When brave browser is set as a default application,
different icon will be used for that file type.
With this PR, `.pdf` and `.svg` files will have different icon.

![image](https://user-images.githubusercontent.com/6786187/121293785-2cf39300-c927-11eb-9868-9cd84ed1de74.png)
![image](https://user-images.githubusercontent.com/6786187/121293834-41379000-c927-11eb-8029-4dcd63757bf9.png)

So far, we set `BraveBHTML` (beta) prog id for supported file types in registry like below.
![image](https://user-images.githubusercontent.com/6786187/121471497-c0e25f00-c9fa-11eb-9aeb-83ba91680448.png)
and that `BraveBHTML` prog id entry has some attributes such as default icon or commands.
That default icon is used for th file type when browser is set as its default application.
default icon of `BraveBHTML`  reg entry is brave browser's default icon.
So, all file types uses same icon.
![image](https://user-images.githubusercontent.com/6786187/121471722-20d90580-c9fb-11eb-8998-7a30c3130ca6.png)

This PR introduces new `BraveFile` prog entry and it uses different icon.
With this PR, new `BraveFile` id is set for `pdf` like below.
![image](https://user-images.githubusercontent.com/6786187/121485468-2a6a6980-ca0b-11eb-90bc-28f712960f6b.png)
and new `BraveFile` entry is newly added.
![image](https://user-images.githubusercontent.com/6786187/121485699-6998ba80-ca0b-11eb-91b8-25ef6a6ecf01.png)



Resolves 

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Download pdf or svg file
2. Set Brave as their default application
3. Check these files' icon is set to brave's file icon(not brave default icon)
4. Uninstall brave and check these files' icon is set properly to other application.
